CraftsmenLtd/coe-batch-q322
# Unit test serverless project

Fixes #

- [x] Depends on other PR being merged first: #400 

## What kind of change does this PR introduce (check at least one)

- [x] Other (unit test)

## How would you describe the changes in a changelog (1-2 lines)?
Unit test cases have been added to test files for the serverless project.
## What was done
- Initialized test environments
- Refactored some parts for unit testing
- Added all event handling test cases
- Injected dependency to test with created mock functions
- Achieved test coverage for the full project by more than 80%.
## How to test
To run all test cases, run these commands from the serverless_lamda directory:
```sh
npm install
npm install -g mocha
npm test
```
To see full test coverage:
```sh
npm install -g nyc
nyc npm test
```


## Deploy risk & follow up

Please assess risk & follow up on deployment of this PR. Elaborate your choice if selecting anything other than 'No risk'.


## Reminder list

- Determine whether REST API Documentation should be updated
- Make an assessment on whether unit tests should be updated and/or added
- Ensure follow up tasks are linked to this PR

QA - if regression testing was requested:

- [ ] QA has verified
